### PR TITLE
Fixed empty user creation

### PR DESCRIPTION
**Pull request by tylerhm**
User's can now leave any field empty in the creation, and our backend will fill it with "No <field>"

### CODE REVIEW

**Comment by suneettipirneni** [comment]
> User's can now leave any field empty in the creation, and our backend will fill it with "No <field>"

I think I would prefer to have the backend store the fields as NULL, mainly because that's standard for representing "absence of". 

Then on the front end we can interpret this result and have the “No field” text

**Comment by tylerhm** [comment]
> > User's can now leave any field empty in the creation, and our backend will fill it with "No "
> 
> I think I would prefer to have the backend store the fields as NULL, mainly because that's standard for representing "absence of".
> 
> Then on the front end we can interpret this result and have the “No field” text

Agreed! Changing now :)

**Comment by tylerhm** [comment]
@suneettipirneni Actually, we would have to do quite a bit of modification for that. The SQL request would have to be dynamic every time we send it. It's not the end of the world, just want to run it by you before I make the modifications

**Comment by suneettipirneni** [comment]
> @suneettipirneni Actually, we would have to do quite a bit of modification for that. The SQL request would have to be dynamic every time we send it. It's not the end of the world, just want to run it by you before I make the modifications

If you’re referring to making the fields null-able in sql that’s fine. The requests *should* stay the same, the php sql statement should interpret null as a SQL NULL value.

**Comment by tylerhm** [comment]
> > @suneettipirneni Actually, we would have to do quite a bit of modification for that. The SQL request would have to be dynamic every time we send it. It's not the end of the world, just want to run it by you before I make the modifications
> 
> If you’re referring to making the fields null-able in sql that’s fine. The requests _should_ stay the same, the php sql statement should interpret null as a SQL NULL value.

So I can just set the field equal to the string "null"?

**Comment by suneettipirneni** [comment]
> > @suneettipirneni Actually, we would have to do quite a bit of modification for that. The SQL request would have to be dynamic every time we send it. It's not the end of the world, just want to run it by you before I make the modifications
> 
> If you’re referring to making the fields null-able in sql that’s fine. The requests _should_ stay the same, the php sql statement should interpret null as a SQL NULL value.

No, modifications aren’t necessary on the backend, because when you save it, it’ll be null (the value). This null value will be interpreted to NULL automatically when the SQL statements is executed. For example:

“null” -> VARCHAR
null -> NULL

**Comment by tylerhm** [comment]
> > > @suneettipirneni Actually, we would have to do quite a bit of modification for that. The SQL request would have to be dynamic every time we send it. It's not the end of the world, just want to run it by you before I make the modifications
> > 
> > 
> > If you’re referring to making the fields null-able in sql that’s fine. The requests _should_ stay the same, the php sql statement should interpret null as a SQL NULL value.
> 
> No, modifications aren’t necessary on the backend, because when you save it, it’ll be null (the value). This null value will be interpreted to NULL automatically when the SQL statements is executed. For example:
> 
> “null” -> VARCHAR
> null -> NULL

Ok, so when filling out the contacts fields how should I approach the the null values? Simply assigning them to null gives a 500 error with the API.

**Comment by tylerhm** [comment]
```
// Get Form Data
	const contact = {};
	data.forEach(function (value, key) {
		if (value === "" || value === "Choose...")
			contact[key] = null;
		else
			contact[key] = value;
	});
```

^^ This gives 500 error when sent accross

**Comment by suneettipirneni** [comment]
> ```
> // Get Form Data
> 	const contact = {};
> 	data.forEach(function (value, key) {
> 		if (value === "" || value === "Choose...")
> 			contact[key] = null;
> 		else
> 			contact[key] = value;
> 	});
> ```
> 
> 
> ^^ This gives 500 error when sent accross

Does your SQL database allow all of the contact fields to be NULL? (except the ID's)

**Comment by tylerhm** [comment]
> > ```
> > // Get Form Data
> > 	const contact = {};
> > 	data.forEach(function (value, key) {
> > 		if (value === "" || value === "Choose...")
> > 			contact[key] = null;
> > 		else
> > 			contact[key] = value;
> > 	});
> > ```
> > 
> > 
> > ^^ This gives 500 error when sent accross
> 
> Does your SQL database allow all of the contact fields to be NULL? (except the ID's)

Ahhh that did it! :man_facepalming: Going to require the user to fill out FirstName and LastName now, and then we should be all set.

**Comment by suneettipirneni** [comment]
> > > ```
> > > // Get Form Data
> > > 	const contact = {};
> > > 	data.forEach(function (value, key) {
> > > 		if (value === "" || value === "Choose...")
> > > 			contact[key] = null;
> > > 		else
> > > 			contact[key] = value;
> > > 	});
> > > ```
> > > 
> > > 
> > > ^^ This gives 500 error when sent accross
> > 
> > 
> > Does your SQL database allow all of the contact fields to be NULL? (except the ID's)
> 
> 
> Ahhh that did it! 🤦‍♂️ Going to require the user to fill out FirstName and LastName now, and then we should be all set.

Awesome!

**Comment by tylerhm** [comment]
Everything should be good to go now :+1:

**Review by suneettipirneni — CHANGES_REQUESTED** [review]
Just one minor thing, but the rest looks good

File: js/contact.js [hunk 1]
```diff
@@ -160,28 +175,32 @@ function displayContact(contactRef) {
...
-		// Skip over not standard views.
-		if (!("contactKey" in editChild.dataset)) {
-			continue;
-		}
+	// Create a functional parser object.
+	const parser = element =>
+		({
+			key: element.dataset.contactKey,
+			targetID: element.dataset.contactTarget
+		});
```
Comment (suneettipirneni) [inline js/contact.js]: This can probably just be a global constant since it's used in more than one function.

**Review by suneettipirneni — CHANGES_REQUESTED** [review]

File: js/contact.js (hunk 1 above)
Comment (tylerhm) [reply js/contact.js]: True! Wasn't quite sure how the whole => thing worked, and if it could be globalized or not.

Comment (suneettipirneni) [reply js/contact.js]: “Arrow Functions” are basically a shorthand/lightweight version of a function.

**Review by suneettipirneni — APPROVED** [review]

**Pull request merged**